lib/charms/grafana_k8s/v0/grafana_metadata.py

```diff
@@ -0,0 +1,250 @@
+    def __init__(self, framework):
+        super().__init__(framework)
+        # Create the GrafanaMetadataRequirer instance, providing the relation name you've used
+        self.grafana_metadata = GrafanaMetadataRequirer(self, "grafana-metadata")
+        self.framework.observe(self.grafana_metadata.on.data_changed, self.do_something_with_metadata)
+```
+
+To access the data elsewhere in the charm, use the provided data accessors.  These return `GrafanaMetadataAppData`
+objects:
+
+```python
+class FooCharm(CharmBase):
+    ...
+    # If using limit=1
+    def do_something_with_metadata(self):
+        # Get exactly one related application's data, raising if more than one is available
+        # note: if not using limit=1, see .get_data_from_all_relations()
+        metadata = self.grafana_metadata.get_data()
+        if metadata is None:
+            self.log("No metadata available yet")
+
+Return:
+        self.log(f"Got Grafana's internal_url: {metadata.internal_url}")
+```
+
+### Provider
+
+To add this relation to your charm as a provider, add the following to your `charmcraft.yaml` or `metadata.yaml`:
+
+```yaml
+provides:
+  grafana-metadata:
+    interface: grafana_metadata
+```
+
+To manage sending data to related applications in your charm, use `GrafanaMetadataProvider`.  Note that
+`GrafanaMetadataProvider` *does not* manage any events, but instead provides a `send_data` method for sending data to
+all related applications.  Triggering `send_data` appropriately is left to the charm author, although generally you want
+to do this at least during relation_joined and leader_elected events.  An example implementation is:
+
+```python
+class FooCharm(CharmBase):
+    def __init__(self, framework):
+        super().__init__(framework)
+        self.grafana_metadata = GrafanaMetadataProvider(
+            charm=self,
+            grafana_uid=self.unique_name,
+            ingress_url=self.external_url,
+            internal_url=self.internal_url,
+            relation_name="grafana-metadata"
+        )
+
+        self.framework.observe(self.on.leader_elected, self.do_something_to_send_data)
+        self.framework.observe(self._charm.on["grafana-metadata"].relation_joined, self.do_something_to_send_data)
+```
+"""
+import json
+import logging
+from typing import List, Optional, Union
+
+
+from ops import CharmBase, BoundEvent, EventBase, CharmEvents, EventSource, Object
+from pydantic import BaseModel, Field
+
+# The unique Charmhub library identifier, never change it
+LIBID = "26290f24974540adb4464b695bd01ea3"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+PYDEPS = ["pydantic>=2"]
+
+log = logging.getLogger(__name__)
+
+DEFAULT_RELATION_NAME = "grafana-info"
+
+
+class GrafanaMetadataAppData(BaseModel):
+    """Data model for the grafana-info interface."""
```

Comment:
```suggestion
    """Data model for the grafana-metadata interface."""
```
do a thorough search for `grafana-info`, i've seen it elsewhere too.